nifi-nar-bundles/nifi-snowflake-bundle/nifi-snowflake-services/src/main/java/org/apache/nifi/snowflake/service/SnowflakeComputingConnectionPool.java

```diff
@@ -0,0 +1,314 @@
+        .fromPropertyDescriptor(DBCPConnectionPool.MAX_WAIT_TIME)
+        .displayName("Max Wait Time")
+        .name("max-wait-time")
+        .build();
+
+    public static final PropertyDescriptor MAX_TOTAL_CONNECTIONS = new PropertyDescriptor.Builder()
+        .fromPropertyDescriptor(DBCPConnectionPool.MAX_TOTAL_CONNECTIONS)
+        .displayName("Max Total Connections")
+        .name("max-total-connections")
+        .build();
+
+    public static final PropertyDescriptor MIN_IDLE = new PropertyDescriptor.Builder()
+        .fromPropertyDescriptor(DBCPConnectionPool.MIN_IDLE)
+        .displayName("Minimum Idle Connections")
+        .name("snowflake-min-idle-conns")
+        .build();
+
+    public static final PropertyDescriptor MAX_IDLE = new PropertyDescriptor.Builder()
+        .fromPropertyDescriptor(DBCPConnectionPool.MAX_IDLE)
+        .displayName("Max Idle Connections")
+        .name("snowflake-max-idle-conns")
+        .build();
+
+    public static final PropertyDescriptor MAX_CONN_LIFETIME = new PropertyDescriptor.Builder()
+        .fromPropertyDescriptor(DBCPConnectionPool.MAX_CONN_LIFETIME)
+        .displayName("Max Connection Lifetime")
+        .name("snowflake-max-conn-lifetime")
+        .build();
+
+    public static final PropertyDescriptor EVICTION_RUN_PERIOD = new PropertyDescriptor.Builder()
+        .fromPropertyDescriptor(DBCPConnectionPool.EVICTION_RUN_PERIOD)
+        .displayName("Time Between Eviction Runs")
+        .name("snowflake-time-between-eviction-runs")
+        .build();
+
+    public static final PropertyDescriptor MIN_EVICTABLE_IDLE_TIME = new PropertyDescriptor.Builder()
+        .fromPropertyDescriptor(DBCPConnectionPool.MIN_EVICTABLE_IDLE_TIME)
+        .displayName("Minimum Evictable Idle Time")
+        .name("snowflake-min-evictable-idle-time")
+        .build();
+
+    public static final PropertyDescriptor SOFT_MIN_EVICTABLE_IDLE_TIME = new PropertyDescriptor.Builder()
+        .fromPropertyDescriptor(DBCPConnectionPool.SOFT_MIN_EVICTABLE_IDLE_TIME)
+        .displayName("Soft Minimum Evictable Idle Time")
+        .name("snowflake-soft-min-evictable-idle-time")
+        .build();
+
+    private static final List<PropertyDescriptor> properties;
+
+    static {
+        final List<PropertyDescriptor> props = new ArrayList<>();
+        props.add(SNOWFLAKE_URL);
+        props.add(SNOWFLAKE_USER);
+        props.add(SNOWFLAKE_PASSWORD);
+        props.add(VALIDATION_QUERY);
+        props.add(MAX_WAIT_TIME);
+        props.add(MAX_TOTAL_CONNECTIONS);
+        props.add(MIN_IDLE);
+        props.add(MAX_IDLE);
+        props.add(MAX_CONN_LIFETIME);
+        props.add(EVICTION_RUN_PERIOD);
+        props.add(MIN_EVICTABLE_IDLE_TIME);
+        props.add(SOFT_MIN_EVICTABLE_IDLE_TIME);
+
+        properties = Collections.unmodifiableList(props);
+    }
+
+    private volatile BasicDataSource dataSource;
```

Comment:
Changing the `DBCPConnectionPool` member `dataSource` to `protected`, or adding an accessor method would avoid having to declare this again.

Reply:
Yeah, making it protected seems to be a good idea.